reactive/kubernetes_worker.py

```diff
@@ -619,9 +620,16 @@ def start_worker():
     creds = db.get('credentials')
     data_changed('kube-control.creds', creds)
 
+    # older kube-control doesn't have has_xcp flag
+    has_xcp = getattr(kube_control, 'has_xcp', False)
+
     create_config(servers[get_unit_number() % len(servers)], creds)
     configure_default_cni(kube_control.get_default_cni())
-    configure_kubelet(dns_domain, dns_ip, registry)
+    try:
+        configure_kubelet(dns_domain, dns_ip, registry, has_xcp=has_xcp)
+    except TypeError:
+        # older kubernetes-common doesn't support the has_xcp flag
+        configure_kubelet(dns_domain, dns_ip, registry)
```

Comment:
this is sad.  maybe `configure_kubelet` should take `**kwargs` so we can stop this kind of nonsense?